### PR TITLE
Adds load function export for storybook__react types

### PR DESCRIPTION
**Pull request by ElliottRoche**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

Based on the technical preview for storybook docs, this load function needs to be included in the types. See page 3 on this google doc: <https://docs.google.com/document/d/1un6YX7xDKEKl5-MVb-egnOYN8dynb5Hf7mq0hipk8JE/edit#>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/storybook__react/index.d.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type RenderFunction = () => Renderable | Renderable[];
 export interface DecoratorParameters {
     [key: string]: any;
 }
-export type StoryDecorator = (story: RenderFunction, context: { kind: string, story: string }) => Renderable | null;
+export type StoryDecorator = (story: RenderFunction, context: { kind: string; story: string }) => Renderable | null;
```
Comment (joscha) [inline types/storybook__react/index.d.ts]: this is a simple object, not an interface, I think we'd keep a comma for these?

File: types/storybook__react/storybook__react-tests.tsx [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+    clearDecorators,
+} from '@storybook/react';
 
 const Decorator = (story: RenderFunction) => <div>{story()}</div>;
 const parameters: DecoratorParameters = { parameter: 'foo' };
 
 forceReRender();
+load(require.context('../src', true, /\.stories\.tsx$/), module);
 
 storiesOf('Welcome', module)
```
Comment (joscha) [inline types/storybook__react/storybook__react-tests.tsx]: everything below here is unrelated, right? Best to remove the noise if it's not needed for the PR to land?

File: types/storybook__react/storybook__react-tests.tsx [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
+    load,
+    DecoratorParameters,
+    clearDecorators,
+} from '@storybook/react';
 
 const Decorator = (story: RenderFunction) => <div>{story()}</div>;
 const parameters: DecoratorParameters = { parameter: 'foo' };
 
 forceReRender();
+load(require.context('../src', true, /\.stories\.tsx$/), module);
```
Comment (joscha) [inline types/storybook__react/storybook__react-tests.tsx]: this looks good to me, however this function is only available in a technical preview right now if I understand the doc correctly that you linked @ElliottRoche?.
I can see it being used here: https://github.com/storybookjs/storybook/blob/454fab82f8be5781a93566f26070596a98bdc9c8/examples/cra-kitchen-sink/.storybook/config.js#L25 so I suspect it is released as some version, but unfortunately, I have no experience on how to publish types for a preview version, do you know?

File: types/storybook__react/index.d.ts [hunk 4]
```diff
@@ -33,6 +33,7 @@ export function setAddon(addon: object): void;
 export function storiesOf(name: string, module: NodeModule): Story;
 export function storiesOf<T>(name: string, module: NodeModule): Story & T;
 export function forceReRender(): void;
+export function load(paths: __WebpackModuleApi.RequireContext, module: NodeModule): void;
```
Comment (joscha) [inline types/storybook__react/index.d.ts]: can we depend on the webpack-env types directly and import `RequireContext`? I just did a short search for `__WebpackModuleApi.RequireContext` in the DT repository and can't seem to find any other usage of this in that way.

File: types/storybook__react/storybook__react-tests.tsx (hunk 2 above)
Comment (ElliottRoche) [reply types/storybook__react/storybook__react-tests.tsx]: Yeah a lot of this is just due to some extension in VS Code doing formatting. I'll pull all those formatting changes out.

File: types/storybook__react/storybook__react-tests.tsx (hunk 3 above)
Comment (ElliottRoche) [reply types/storybook__react/storybook__react-tests.tsx]: Yeah this for the alpha version of storybook, currently v5.2.0-alpha.38, that introduces Storybook docs as a replacement or enhancement for a lot of the addon packages like addon-info.

No idea how to publish types for a preview, but this load function didn't appear to affect anything from previous storybook versions so I figured I'd go ahead and add it.

File: types/storybook__react/index.d.ts (hunk 4 above)
Comment (ElliottRoche) [reply types/storybook__react/index.d.ts]: As far as I could tell, RequireContext isn't exported by webpack anywhere and I couldn't get a direct import working, but if you know of a different way I'm happy to switch it over!

File: types/storybook__react/index.d.ts (hunk 1 above)
Comment (ElliottRoche) [reply types/storybook__react/index.d.ts]: More auto formatting that I can remove.

Comment (ElliottRoche) [reply types/storybook__react/index.d.ts]: Actually this appears to be happening from a husky pre-commit hook that's part of the DefinitelyTyped package.json file.

Screen Shot 2019-07-09 at 10 29 03 AM

File: types/storybook__react/storybook__react-tests.tsx (hunk 2 above)
Comment (ElliottRoche) [reply types/storybook__react/storybook__react-tests.tsx]: This is also coming from that same pre-commit hook mentioned above.

**Comment by ElliottRoche** [comment]
Storybooks types are being moved to the storybook repo soon, so I'm going to close this PR in anticipation of that happening.

**Pull request closed without merging**